src/client.c

```diff
@@ -221,8 +221,16 @@ static bool login_via_cert(PgSocket *client)
 		goto fail;
 
 	log_debug("TLS cert login: %s", tls_peer_cert_subject(client->sbuf.tls));
-	if (!tls_peer_cert_contains_name(client->sbuf.tls, client->login_user->name)) {
-		slog_error(client, "TLS certificate name mismatch");
+
+	if (rule && rule->identmap) {
+		if (!tls_peer_cert_contains_name(client->sbuf.tls, rule->identmap->system_user_name))
```

Comment:
We should log errors here, failures in these kind of things are hard enough to debug already.